api/parts/data/dataBatchReader.js

```diff
@@ -0,0 +1,200 @@
+ */
+class dataBatchReader {
+    /** 
+    * @param {Object} db - Database object
+    * @param {Object} options - Options object
+    * @param {function} onData - Finction to call when getting new data from stream
+    */
+    constructor(db, options, onData,) {
+        log.e("Creating dataBatchReader", JSON.stringify(options));
+        this.db = db;
+        this.pipeline = options.pipeline || [];
+        this.match = options.match || {};
+        this.timefield = options.timefield || "cd"; //default time field is cd
+        this.lastToken = null;
+        this.name = options.name || "";
+        this.collection = options.collection || "drill_events";
+        this.options = options.options;
+        this.onData = onData;
+        this.interval = options.interval || 10000;
+        if (plugins.getConfig("aggregator") && plugins.getConfig("aggregator").interval) {
+            log.e("Using aggregator config interval for dataBatchReader", plugins.getConfig("aggregator").interval);
+            this.interval = plugins.getConfig("aggregator").interval;
+        }
+        else {
+            log.e("No interval set. Using default " + this.interval);
+        }
+        this.reviveInterval = options.reviveInterval || 60000; //1 minute
+        this.setUp();
+
+        this.localId = this.name + Date.now().valueOf();
+
+        this.intervalRunner = setInterval(this.checkState.bind(this), this.reviveInterval);
+    }
+
+    /** 
+    * Check if stream is closed and restart if needed
+    */
+    async checkState() {
+        log.e("Checking state for batcher to revive if dead");
+        //Check last token in database. If it is older than 60 seconds. Set up again
+        try {
+            var token = await common.db.collection("plugins").findOne({"_id": "_changeStreams"}, {projection: {[this.name]: 1}});
+            if (token && token[this.name] && token[this.name].lu) {
+                var lu = token[this.name].lu.valueOf();
+                var now = Date.now().valueOf();
+                if (now - lu > this.reviveInterval) {
+                    log.e("Stream is closed. Restarting it");
+                    this.setUp();
+                }
+            }
+            else {
+                log.e("Processor does not exists. Set it up again.");
+                this.setUp();
+            }
+        }
+        catch (err) {
+            log.e("Error checking state for batcher", err);
+        }
+    }
+
+    /**
+     * Processes range of dates
+     * @param {date} cd  - start time
+     */
+    async processNextDateRange() {
+        var lastToken = await common.db.collection("plugins").findOne({"_id": "_changeStreams"}, {projection: {[this.name]: 1}});
+        var cd = lastToken && lastToken[this.name] ? lastToken[this.name].cd : Date.now();
+        var cd2 = cd.valueOf() + this.interval;
+        var now = Date.now().valueOf();
+        while (cd2 < now) {
+            cd2 = new Date(cd2);
```

Comment:
cd2 is first defined as a number, then assigned to a date object.
Also cd2 is a bad naming, doesnt tell me why we needed second cd or why it is being assigned different type